packages/toast/toast.js

```diff
@@ -0,0 +1,143 @@
+import { LitElement, css, html } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { when } from 'lit/directives/when.js';
+import { toast as c } from '@fabric-ds/component-classes';
+import { expand, collapse } from 'element-collapse';
+import { closeSVG, successSVG, failureSVG } from './svgs';
+
+export class FabricToast extends LitElement {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  static properties = {
+    id: { type: String, attribute: true, reflect: true },
+    type: { type: String, attribute: true, reflect: true },
+    text: { type: String, attribute: true, reflect: true },
+    canclose: { type: Boolean, attribute: true, reflect: true },
+  };
+
+  constructor() {
+    super();
+    this.id = Date.now().toString(36) + Math.random().toString(36).slice(2, 5);
+    this.type = 'success';
+    this.text = '';
+    this.canclose = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+  }
+
+  updated() {
+    if (!this._expanded && this._wrapper) expand(this._wrapper, () => (this._expanded = true));
+  }
+
+  get _primaryClasses() {
+    return classMap({
+      [c.toast]: true,
+      [c.toastPositive]: this.type === 'success',
+      [c.toastWarning]: this.type === 'warning',
+      [c.toastNegative]: this.type === 'error',
+      [c.toastNeutral]: this.type === 'info',
+    });
+  }
+
+  get _iconClasses() {
+    return classMap({
+      [c.toastIcon]: true,
+      [c.toastIconPositive]: this.type == 'success',
+      [c.toastIconWarning]: this.type === 'warning',
+      [c.toastIconNegative]: this.type === 'error',
+      [c.toastIconNeutral]: this.type === 'info',
+    });
+  }
+
+  get _wrapper() {
+    return this.renderRoot?.querySelector(`section`) ?? null;
+  }
+
+  get _success() {
+    return this.type === 'success';
+  }
+
+  get _warning() {
+    return this.type === 'warning';
+  }
+
+  get _error() {
+    return this.type === 'error';
+  }
+
+  get _info() {
+    return this.type === 'info';
+  }
+
+  get _role() {
+    return this._error || this._warning ? 'alert' : 'status';
+  }
+
+  get _typeLabel() {
+    if (this._success) return 'Vellykket';
+    if (this._error) return 'Feil';
+    if (this._warning) return 'Varsel';
+    return 'Info';
+  }
+
+  get _iconMarkup() {
+    return this._success
+      ? successSVG({ typeLabel: this._typeLabel })
+      : failureSVG({ typeLabel: this.typeLabel, isInfo: this._info });
+  }
+
+  async collapse() {
+    return new Promise((resolve) => {
+      if (this._expanded && this._wrapper) collapse(this._wrapper, resolve);
+      else resolve();
+    });
+  }
+
+  close() {
+    const event = new CustomEvent('close', {
+      detail: { id: this.id },
+      bubbles: true,
+      composed: true,
+    });
+    this.updateComplete.then(() => this.dispatchEvent(event));
```

Comment:
A `close` event is triggered whenever the user clicks the x button on the toast. This even works in line with standard dom event listeners so in order to listen for this event you need to use `addEventListener` like so:

```js
const el = document.querySelector('f-toast')
el.addEventListener('close', (event) => { ... })
```

When using Lit Element you get some syntactic sugar like so:

```js
html`<f-toast ... @close=${(event) => { ... }}></f-toast>`;
```